src/Prefix.jl

```diff
@@ -338,13 +348,23 @@ function setup_workspace(build_path::AbstractString, sources::Vector{<:SetupSour
     wrapperdir = joinpath(workspace, "compiler_wrappers")
     mkdir.((srcdir, destdir, metadir))
 
+    if srcdir_overlay
+        mkdir(joinpath(build_path, nonce, "srcdir_work"))
+        srcdir = joinpath(build_path, nonce, "srcdir_lower")
+        mkdir(srcdir)
+    end
```

Comment:
I would prefer that this is always `srcdir`, not `srcdir_lower`; reduces complexity and makes it so I don't have to think about what's going on.  We can bindmount it to `${WORKSPACE}/srcdir_lower` just before we mount the overlay to get this to work.